docs/design/ltc_disc.md

```diff
@@ -0,0 +1,134 @@
+# Design Doc: Accelerate PyTorch training vis BladeDISC and PyTorch Lazy Tensor Core
+
+BladeDISC is an end-to-end compiler that accelerates AI workload based on MLIR.
+PyTorch [Lazy Tensor Core (LTC)](https://github.com/pytorch/pytorch/blob/lazy_tensor_staging/lazy_tensor_core/README.md)
+provides a infrastructure to connect to PyTorch to deep learning accelerators
+especial on training.  This design document aims to introduce how to extend
+PyTorch LTC to accelerate PyTorch training via BladeDISC.
+
+## Overview
+
+In PyTorch LTC world, TorchScript backend lowing Lazy Tensors into TorchScript
+and execute it with `torch::jit:GraphExecutor`.  LTC provides a way to add a
+new vender with implementing `BackendImplInterface`.  To BladeDISC, the ROI
+is too low to lowing all Lazy Tensors to BladeDISC IR, a better way is to extend
+TorchScript backend, which only optimizes a sub-graph of the whole TorchScript graph.
+
+![overview](./images/ltc_disc_overview.png)
+
+The above illustration shows the key process:
+
+1. traverse the whole graph and cluster the Disc compilable nodes into a
+   sub-graph.
+1. convert TorchScript IR into mhlo IR.
+1. call the Disc compiling pipeline into an executable program.
+1. implement Disc `CustomClassHolder` to bind the executable program above.
+1. replace the disc cluster sub-graph on the origin graph with
+`prim::CallMethod` symbol to call Disc executable at runtime.
+
+## Implementation
+
+### Extend LTC TorchScript Backend
```

Comment:
FYI. We now depend on unstable API and source codes of PyTorch LazyTensorCore. I have created an issue to track it https://github.com/alibaba/BladeDISC/issues/239

Reply:
I brought this up because I think it will be important to any work on LTC backend, not because it is an issue that should be addressed here in BladeDISC. @penguinwu has a nice write-up about the current status for PyTorch graph capture [here](https://dev-discuss.pytorch.org/t/the-nuances-of-pytorch-graph-capture/501).